### PR TITLE
Initial work on action

### PR DESCRIPTION
**Pull request by paulolieuthier**

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      /go/pkg/mod
+
+  key:
+    description: 'An explicit key for restoring and saving the cache'
+    default: ${{ runner.os }}-go-dependencies-${{ hashFiles('go.sum') }}
+
+  restore-keys:
+    description: 'An ordered list of keys to use for restoring the cache if no cache hit occurred for key'
+    default: |
+      ${{ runner.os }}-go-dependencies-
```
Comment (t0rr3sp3dr0) [inline action.yml:25]: I don't think we should make this configurable. This is an opinionated setup for Go, if the user disables the cache, it's better to just use actions/setup-go instead.

File: action.yml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    run: |
+      UNFORMATED=$(gofmt -l ${{ inputs.source-path }})
+      COUNT=$(echo $UNFORMATED | wc -w)
+      if [ $COUNT -eq 0 ]; then
+          echo "Files formatted properly!"
+          exit 0
+      fi
+      echo "The following files are not properly formatted:"
+      echo >&2 $UNFORMATED
+      exit 1
```
Comment (t0rr3sp3dr0) [inline action.yml:67]: This should not be in the setup action. We can create another action that performs various checks for Go, Java, and other languages. But this is out of scope for a setup action.

**Review by t0rr3sp3dr0 — CHANGES_REQUESTED** [review]

File: action.yml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+  - name: Install GCC
+    shell: bash
+    if: inputs.gcc
+    run: sudo apt update && sudo apt install -y gcc
```
[the remark this answers is not in the document]
Comment (paulolieuthier) [reply action.yml:72]: I've seen projects that only need `gcc` as an extra package.

> We could have an input as part of the bootstrap action to install packages listed on the input, where the user could list gcc and other dependencies required for the build.

Great idea.

File: action.yml (hunk 2 above)
Comment (paulolieuthier) [reply action.yml:67]: Fair enough.

File: action.yml (hunk 1 above)
Comment (paulolieuthier) [reply action.yml:25]: Fair enough.

File: action.yml (hunk 2 above)
Comment (paulolieuthier) [reply action.yml:67]: The thing is that of the languages we use, I think only Go has a default formatting tool. incognia-backend, for instance, uses a Gradle plugin for that.

Comment (t0rr3sp3dr0) [reply action.yml:67]: Python algo has a well-defined formatting style: https://www.python.org/dev/peps/pep-0008/

Comment (t0rr3sp3dr0) [reply action.yml:67]: I would be also to incorporate this tool across the company

File: action.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+runs:
+  using: composite
+  steps:
+  - name: Setup Go proper
+    uses: actions/setup-go@v2
+    with:
+      go-version: ${{ inputs.go-version }}
+
+  - name: Cache Go dependencies
+    if: inputs.cache
```
Comment (t0rr3sp3dr0) [inline action.yml:18]: You forgot to remove this line

**Review by t0rr3sp3dr0 — APPROVED** [review]

**Pull request merged**